test/lib/bamboo/adapters/test_adapter_test.exs

```diff
@@ -172,6 +172,69 @@ defmodule Bamboo.TestAdapterTest do
     end
   end
 
+  test "refute_email_delivered_with when email does not match" do
+    mail =
+      new_email(
+        to: to = [{nil, "foo@bar.com"}],
+        from: {nil, "baz@bar.com"},
+        subject: subject = "coffee"
+      )
+
+    TestMailer.deliver_now(mail)
+    refute_email_delivered_with(subject: ~r/tea/)
+
+    TestMailer.deliver_now(mail)
+
+    try do
+      refute_email_delivered_with(subject: subject)
```

Comment:
This, and the others below, are in the test called `when email does not match` but it seems they are testing `when email matches`, which there is a separate test for? Thoughts on regrouping them, or alternatively removing some of them if they don't each add much additional value?

Also, if we do keep all of them, thoughts on extracting a helper function for the repeated code?

Reply:
Right, I don't think they add too much value. I've refactored the tests a bit if you want to take another look.

Reply:
Nice, thanks so much for this!